### ISSUE TITLE
Check WCAG compliance

### ISSUE THREAD

**Issue by Danwhy** [dwyl/autoform#3]
We need to check if the Phoenix forms we're currently rendering are fully WCAG compliant, and if not implement the necessary changes

https://www.w3.org/WAI/tutorials/forms/

**Comment by Danwhy** [comment]
What we have so far passes all accessibility checks for basic 'text' and 'number' type inputs, but as the forms grow in size and complexity there are some things we'll have to take into account to make sure they remain compliant:

- __Labeling__: We currently include a label for every input, which is the name of the field. In the future, we may want to allow users to determine whether or not they want to display a label. If we do this, we'll have to make sure to include an aria label on the inputs to ensure they remain accessible.
- __Grouping__: We currently have no way of structuring related data in a form, whether that would be grouping multiple schemas into one form (addresses, other user details), or checkbox and radio buttons. As and when we include this functionality, we'll have to ensure we're grouping related together using `fieldsets`.
- __Instructions__: When we're providing instructions, such as a specific format for a date, we need to make sure we're displaying the instructions in an accessible way. We can either use labels, or aria attributes for this.
- __Validation__: When we're validating data, we need to make sure we're using the `required` and `aria-required` attributes for required fields, as well as all available html5 validations.

**Comment by nelsonic** [comment]
@Danwhy _how_ can we _automate_ checking the ARIA/WCAG compliance of the auto-generated forms? 
e.g: can/should we add https://webhint.io checks during the CI of this repo? 🤔

**Comment by Danwhy** [comment]
We'd need to do some work to get a server running during our tests using a webdriver, but it should be doable, whether it's using webhint, or something else like [pa11y](https://github.com/pa11y/pa11y-ci)

**Comment by nelsonic** [comment]
@Danwhy agreed. not too concerned which checking/CI tool we use (_the faster the better_).
Does _automated_ checking require a _separate_ issue?
Or can it be added as a checklist item to _this_ issue?

**Comment by Danwhy** [comment]
I think it could be part of a separate issue around setting up CI in general. I'll create one for it

**Comment by Danwhy** [comment]
Accessibility checking in CI set up in #6
